R/ard_complex.R

```diff
@@ -78,6 +78,8 @@ ard_complex.data.frame <- function(data,
   process_selectors(data, variables = {{ variables }})
```

Comment:
Hi @ddsjoberg, are we missing `by = {{ by }}, strata = {{strata}}` in this call to `process_selectors`? Just noticed as it means `data[c(by, strata)]` only works when `by` and `strata` are character vectors currently but appreciate there may be another reason for it as this is what's specified in the documentation for `ard_complex()`.

Reply:
Ahhh, good points! I took a look at the function, and it looks like the by and strata args are being passed directly to `ard_continuous()` using the embracing brackets (`{{ }}`). So these will be checked there and we can skip the check here.

Reply:
Thanks @ddsjoberg! I've removed the factor checks from `ard_complex()` 😃